spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/core/ScheduleRequest.java

```diff
@@ -0,0 +1,121 @@
+ * and any deployment properties.
+ *
+ * Deployment properties are related to a specific implementation of the SPI
+ * and will never be passed into an app itself. For example, a runtime container
+ * may allow the definition of various settings for a context where the actual
+ * app is executed, such as allowed memory, cpu or simply a way to define
+ * collocation like node labeling.
+ *
+ * For passing properties into the app itself, use {@link AppDefinition#getProperties()}.
+ * Those could be passed as env vars, or whatever approach is best for the target
+ * platform. Each deployer implementation should clearly document how it handles
+ * these properties.
+ *
+ * For passing command line arguments into the app itself, use {@link #commandlineArguments}.
+ *
+ * @author Glenn Renfro
+ */
+public class ScheduleRequest extends AppDeploymentRequest{
+
+	/**
+	 * The name of the Schedule.
+	 */
+	private final String scheduleName;
+
+	private Map<String, String> schedulerProperties;
+
+	/**
+	 * Construct an {@code AppDeploymentRequest}.
+	 *
+	 * @param definition app definition.
+	 * @param schedulerProperties properties that contain scheduler specific informaton.
+	 * @param deploymentProperties map of deployment properties; may be {@code null}.
+	 * @param scheduleName the name associated with the schedule.
+	 */
+	public ScheduleRequest(AppDefinition definition, Map<String, String> schedulerProperties,
+			Map<String, String> deploymentProperties, String scheduleName, Resource resource) {
+		this(definition, schedulerProperties, deploymentProperties, null, scheduleName, resource);
+	}
+
+	/**
+	 * Construct an {@code AppDeploymentRequest}.
+	 *
+	 * @param definition app definition
+	 * @param schedulerProperties properties that contain scheduler specific informaton.
+	 * @param deploymentProperties map of deployment properties; may be {@code null}
+	 * @param commandlineArguments set of command line arguments; may be {@code null}
+	 * @param scheduleName the name associated with the schedule.
+	 */
+	public ScheduleRequest(AppDefinition definition,  Map<String, String> schedulerProperties,
+			Map<String, String> deploymentProperties, List<String> commandlineArguments,
+			String scheduleName, Resource resource) {
+		super(definition, resource, deploymentProperties, commandlineArguments);
+		this.scheduleName = scheduleName;
+		this.schedulerProperties = schedulerProperties == null
+				? Collections.<String, String>emptyMap()
+				: Collections.unmodifiableMap(schedulerProperties);
+	}
+
+	/**
+	 * Construct an {@code AppDeploymentRequest} with no deployment properties.
+	 *
+	 * @param definition app definition
+ 	 * @param schedulerProperties properties that contain scheduler specific informaton.
+	 * @param scheduleName the name associated with the schedule.
+	 */
+	public ScheduleRequest(AppDefinition definition, Map<String, String> schedulerProperties, String scheduleName, Resource resource) {
```

Comment:
I think we can get rid of this constructor.  We don't have such a constructor in AppDeploymentRequest (without deployer properties) so not sure why we need it here.